src/test/java/tdl/datapoint/video/VideoDatapointAcceptanceTest.java

```diff
@@ -113,28 +122,89 @@ public void tearDown() throws Exception {
+        //TODO compare videos using dev-screen-record's logic: see acceptance test that generates and reads QRcode
+        assertThatFilesAreEqual("Expect the files to match in content", actualAccumulatorVideo, expectedAccumulatorVideo);
+    }
+
+    @Test
+    public void upload_second_video() throws Exception {
+        // Given - The participant produces Video files while solving a challenge
+        String challengeId = "TCH";
+        String participantId = generateId();
+        String s3AccumulatorVideoDestination = String.format("%s/%s/real-recording.mp4", challengeId, participantId);
+        TestVideoFile accumulatorVideo = new TestVideoFile("tdl/datapoint/video/second_video_upload/before/real-recording.mp4");
+        String s3SecondVideoDestination = String.format("%s/%s/screencast_2.mp4", challengeId, participantId);
+        TestVideoFile s3SecondVideo = new TestVideoFile("screencast_20180727T225445.mp4");
+
+        // When - Upload event happens
+        localS3AccumulatedVideoBucket.putObject(accumulatorVideo.asFile(), s3AccumulatorVideoDestination);
+        S3Event s3Event = localS3SplitVideosBucket.putObject(s3SecondVideo.asFile(), s3SecondVideoDestination);
+        videoUploadHandler.handleRequest(
+                convertToMap(wrapAsSNSEvent(s3Event)),
+                NO_CONTEXT);
+
+        waitForQueueToReceiveEvents();
+
+        // Then - Raw video uploaded events are computed for the deploy tags
+        assertThat("Raw video update events match check: 1 events expected", rawVideoUpdatedEvents.size(), equalTo(1));
+        System.out.println("Received video events: " + rawVideoUpdatedEvents);
+        rawVideoUpdatedEvents.sort(Comparator.comparing(RawVideoUpdatedEvent::getChallengeId));
+        RawVideoUpdatedEvent rawVideoUploaded = rawVideoUpdatedEvents.get(FIRST_ACCUMULATED_VIDEO_EVENT);
+        assertThat("participantId matching", rawVideoUploaded.getParticipant(), equalTo(participantId));
+        assertThat("challengeId matching", rawVideoUploaded.getChallengeId(), equalTo(challengeId));
+        String key = String.format("%s/%s/real-recording.mp4", challengeId, participantId);
+        //TODO check minio to see if we can get a http url & aws s3 - see docs
+        assertThat("Video link match check: expecting something like s3://tdl-official-videos/.../real-recording.mp4", rawVideoUploaded.getVideoLink(), equalTo(
+                String.format("s3://tdl-official-videos/%s", key))
+        );
+
+        Path expectedAccumulatorVideo = new TestVideoFile("tdl/datapoint/video/second_video_upload/after/real-recording.mp4").asFile().toPath();
+        //TODO use Http instead of s3
+        Path actualAccumulatorVideo = new TestVideoFile(localS3AccumulatedVideoBucket, key).getS3Object("real-recording", ".mp4").toPath();
+        //TODO compare videos using dev-screen-record's logic: see acceptance test that generates and reads QRcode
```

Comment:
I have managed to locate the piece of code I was telling you about:
- Videos with barcode: https://github.com/julianghionoiu/dev-screen-anonymise/blob/master/src/test/resources/rec_barcode_matrix/recording_from_matrix_of_barcodes_at_4x.mp4
- how to generate one: https://github.com/julianghionoiu/dev-screen-anonymise/blob/master/src/test/java/tdl/record/image/input/GenerateInputWithMatrixOfBarcodes.java
- how to read barcodes: https://github.com/julianghionoiu/dev-screen-anonymise/blob/master/src/test/resources/rec_barcode_matrix/recording_from_matrix_of_barcodes_at_4x.mp4

Reply:
Will look into it asap

Reply:
In our case, that needs to be a one barcode recording. The functionality needs to be extracted in a separate test library anyway